clientcontroller/opstackl2/consumer.go

```diff
@@ -264,13 +271,31 @@ func (cc *OPStackL2ConsumerController) SubmitBatchFinalitySigs(
 }
 
 // QueryFinalityProviderHasPower queries whether the finality provider has voting power at a given height
-// This interface function only used for checking if the FP is eligible for submitting sigs.
-// Now we can simply hardcode the voting power to true
-// TODO: see this issue https://github.com/babylonlabs-io/finality-provider/issues/390 for more details
 func (cc *OPStackL2ConsumerController) QueryFinalityProviderHasPower(fpPk *btcec.PublicKey, blockHeight uint64) (bool, error) {
+	pubRand, err := cc.QueryLastPublicRandCommit(fpPk)
+	if err != nil {
+		return false, err
+	}
+	if pubRand == nil {
+		return false, nil
+	}
+
+	// fp has 0 voting power if there is no public randomness at this height
+	lastCommittedPubRandHeight := pubRand.StartHeight + pubRand.NumPubRand - 1
```

Comment:
Nit: `pubRand.StartHeight + pubRand.NumPubRand - 1` can be a function `EndHeight()` under `pubRand`